refactor(robot-server, app): wire up tip length calibration session details and commands

b-cooper opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

Wire up the tip length calibration session with the run app. Fills out session details that are returned from robot-server, and replaces front end stubs of session data with references to real session.

# Changelog


- Add contents to the session details returned from the robot-server upon performing an action within
a tip length calibration session.

- Replace the stubbed out calls to commands and session in the run app, with actual calls and references to the created tip length calibration session through sessions api machinery in redux.

- Further divide the session concerns of the tip length calibration from the page component
within which it is rendered.

- Add unit tests for all implemented child components of the TipLengthCalibration view layer
# Review requests

- [ ] Start a tip length calibration session through the run app (This requires the "enable Tip Length Cal" feature flag to be flipped). You should now be able to step through the entire flow by clicking through the main CTA buttons at the bottom of each screen. (Note only happy path wired up, not all hw commands will occur) 

# Risk assessment

nearly all of this is still behind the "enable Tip Length Cal" feature flag, the only existing code that this touches is adding a new tested utility method to `geometry.Deck` for accessing all non-fixture slots.

b-cooper committed:
create session details for tip length calibration

b-cooper committed:
rename component file in start case

b-cooper committed:
wire up calibration intro screen to new session details

b-cooper committed:
wire up send command and start session

b-cooper committed:
remove lint

amitlissack commented on robot-server/robot_server/robot/calibration/tip_length/models.py:
```diff
@@ -0,0 +1,85 @@
+from pydantic import BaseModel, Field
+
+
+# TODO: BC: the mount field here is typed as a string
```
What are the serialization problems? Are they string enums?

b-cooper replied:
This comment is copied over from the cal check models where the `mount` enum was being serialized as the enum index as opposed to the value.  As this draft PR is still a work in progress, if you know of a way to fix it in place, I'll add it in.

amitlissack commented on robot-server/robot_server/robot/calibration/tip_length/user_flow.py:
```diff
@@ -74,6 +78,34 @@ def __init__(self,
+        return self._current_state
+
+    @property
+    def pipette(self) -> AttachedPipette:
```
Minor point really, but I would make this a function and not a property. Same as `required_labware` below. 

I might be in the minority on this, but I feel like a property should barely do more than return a member variable. These are methods that create and return new. They're not really properties.

b-cooper replied:
That makes sense to me. Will make the change now.
In addition, expect some more changes on this PR as it's a WIP.

b-cooper committed:
avoid complex properties

b-cooper committed:
confirm exit

b-cooper committed:
add unit tests for components

b-cooper committed:
refactor(robot-server, app): wire up tip length calibration session details and commands

Add contents to the session details returned from the robot-server upon performing an action within
a tip length calibration session. Replace the stubbed out calls to commands and session in the run
app, with actual calls and references to the created session through sessions api machinery in
redux. Further divide the session concerns of the tip length calibration from the page component
within which it is rendered.

b-cooper committed:
add move to tip rack identity transition for preparingPipette state

amitlissack commented on robot-server/robot_server/robot/calibration/tip_length/models.py:
```diff
@@ -0,0 +1,85 @@
+    name: Optional[str] =\
+        Field(None, description="Short name of pipette model without"
+                                "generation version")
+    tip_length: Optional[float] =\
```
`tipLength`. We are doing camel case in http bodies..

b-cooper replied:
Good catch, don't know how that slipped back in there.

sfoster1 commented on robot-server/robot_server/robot/calibration/tip_length/models.py:
```diff
@@ -0,0 +1,85 @@
+# the middle ware before they are returned to the client
+class AttachedPipette(BaseModel):
+    """Pipette (if any) attached to the mount"""
+    model: Optional[str] =\
```
Do these need to be optional? Tip length calibration can only be invoked on a pipette if it is present. We should never have a non-attached pipette,  and that should be enforced in the model.

sfoster1 reviewed (changes requested):
I think we should make the `AttachedPipette` values non-optional (if tip length cal is running, there should always be a pipette).

b-cooper committed:
camel case and non-optional

b-cooper committed:
lint fixups

b-cooper committed:
merge in create params changes

sfoster1 reviewed (approved):
Python side LGTM

ahiuchingau commented on app/src/pages/Calibrate/CalibrateTipLengthControl.js:
```diff
@@ -0,0 +1,141 @@
+    <>
+      <CalibrationInfoBox
+        confirmed={hasCalibrated}
+        title={`${mount} pipette calibration`}
```
but this can also totally be in a separate PR!

b-cooper replied:
Good call.  Let's add this into #6155 before when we rebase it with these changes.
